### PR TITLE
Fixed bug in CommandRequirements and BasicCommandRequirements 

### PR DESCRIPTION
**Pull request by DenuxPlays**
If the methods in those two classes would be given null as an argument.

### CODE REVIEW

File: src/main/java/com/dynxsty/dih4jda/InteractionHandler.java [hunk 1]
```diff
@@ -111,6 +111,10 @@ protected InteractionHandler(@Nonnull DIH4JDA dih4jda) {
 				findSlashCommands(pkg);
 				findContextCommands(pkg);
 			} catch (ReflectiveOperationException | DIH4JDAException e) {
+				if (e.getCause() instanceof IllegalArgumentException) {
+					DIH4JDALogger.warn("Failed to load requirements in package %s: %s", pkg, e.getCause().getMessage());
+					continue;
+				}
```
Comment (DenuxPlays) [inline src/main/java/com/dynxsty/dih4jda/InteractionHandler.java:117]: I am not quite sure if we should continue the loop or not

**Review by MoonTM-GIT — APPROVED** [review]

**Comment by DynxstyGIT** [comment]
Please improve the wording of the Exception Messages, so it doesn't sound like a first-grader wrote them.

**Comment by DenuxPlays** [comment]
> Please improve the wording of the Exception Messages, so it doesn't sound like a first-grader wrote them.

GitHub Copilot wrote them lmao

File: src/main/java/com/dynxsty/dih4jda/InteractionHandler.java [hunk 2]
```diff
@@ -111,7 +111,15 @@ protected InteractionHandler(@Nonnull DIH4JDA dih4jda) {
...
 			} catch (ReflectiveOperationException | DIH4JDAException e) {
-				DIH4JDALogger.error("An error occurred while initializing commands in package %s: %s", pkg, e.getMessage());
+				if (e.getCause() instanceof IllegalArgumentException) {
+					DIH4JDALogger.warn("Failed to load requirements in package %s: %s", pkg, e.getCause().getMessage());
+					continue;
+				}
+				String message = e.getMessage();
+				if (message == null) {
+					message = e.getCause().getMessage();
+				}
```
Comment (DynxstyGIT) [inline src/main/java/com/dynxsty/dih4jda/InteractionHandler.java:121]: And what the flip is this?? Just do 
```suggestion
				String message = e.getMessage() == null ? e.getCause().getMessage() : e.getMessage();
```

**Review by DynxstyGIT — CHANGES_REQUESTED** [review]

**Comment by DynxstyGIT** [comment]
And what even is the actual fix? Throwing an exception with the arrays are null?

**Comment by DenuxPlays** [comment]
> And what even is the actual fix? Throwing an exception when the arrays are null?

better than your error message that was "null"

**Comment by DynxstyGIT** [comment]
What??

**Comment by DenuxPlays** [comment]
> What??

image
Your error message

File: src/main/java/com/dynxsty/dih4jda/interactions/commands/BaseCommandRequirements.java [hunk 3]
```diff
@@ -28,6 +28,9 @@ public abstract class BaseCommandRequirements extends CommandRequirements {
 	 * @param whitelisted An array of {@link Long}s.
 	 */
 	public final void whitelistGuilds(Long... whitelisted) {
+		if (whitelisted == null) {
+			throw new IllegalArgumentException("whitelisted cannot be null");
+		}
 		if (type != RegistrationType.GUILD) {
 			throw new UnsupportedOperationException("Cannot whitelist Guilds for Global Commands!");
 		}
```
Comment (DynxstyGIT) [inline src/main/java/com/dynxsty/dih4jda/interactions/commands/BaseCommandRequirements.java:36]: Check first for the type, then if the array is null

**Review by DynxstyGIT — CHANGES_REQUESTED** [review]

**Comment by DynxstyGIT** [comment]
> > What??
> 
> image Your error message

Bro thats the default NPE message...

**Comment by DenuxPlays** [comment]
> > > What??
> > 
> > 
> > image Your error message
> 
> Bro thats the default NPE message...

but still nobody know what the hell is wrong

File: src/main/java/com/dynxsty/dih4jda/interactions/commands/BaseCommandRequirements.java (hunk 3 above)
Comment (DenuxPlays) [reply src/main/java/com/dynxsty/dih4jda/interactions/commands/BaseCommandRequirements.java:36]: but why? if the array is null you dont have to check for the type

**Comment by DenuxPlays** [comment]
Irrelevant due to: #50

**Pull request closed without merging**